spec/inputs/kafka_spec.rb

```diff
@@ -22,11 +23,71 @@ def initialize(topic, partition, key, message, offset)
   end
 end
 
+class TestMessageStreamIterator < Queue
+  def initialize(stopper)
+    super()
+    @shutdown_called = stopper
+  end
+
+  def hasNext
+    !self.empty?
+  end
+
+  def next
+    self.pop
+  end
+end
+
+class TestInfiniteStreamIterator < TestMessageStreamIterator
+  def initialize(stopper)
+    super(stopper)
+  end
+
+  def hasNext
+    unless @shutdown_called.value
+      blah = TestMessageAndMetadata.new(@topic, 0, nil, 'Kafka message', 1)
+      self << blah
+    end
+    super
+  end
+end
+
+class TestExceptionStreamIterator < TestMessageStreamIterator
+  def initialize(stopper)
+    super(stopper)
+  end
+
+  def hasNext
+    raise 'Exception from TestExceptionStreamIterator'
```

Comment:
Since exceptions within threads are delicate in the Logstash environment. maybe we should test 
with a real exception that is thrown from the `hasNext`[1]?

[1] https://github.com/apache/kafka/blob/0.8.2/core/src/main/scala/kafka/utils/IteratorTemplate.scala#L52